data/source/MongoDb.php

```diff
@@ -425,9 +413,12 @@ public function disconnect() {
 	 * @return array Returns an array of objects to which models can connect.
 	 */
 	public function sources($class = null) {
-		$this->_checkConnection();
-		$conn = $this->connection;
-		return array_map(function($col) { return $col->getName(); }, $conn->listCollections());
+		$collections = $this->server->executeCommand($this->_config['database'], new Command(['listCollections' => 1]));
+		$names = [];
+		foreach ($collections as $collection) {
+			$names[] = $collection->name;
+		}
+		return $names;
```

Comment:
Can't array_map (which looks nicer to me) be used because `$collections` isn't an array?

Reply:
I tried several versions of array_map, but since the return value of MongoDB can be varying this is actually the nicest (read: most readable) way of doing it. I say that as someone who prefers to use array_map for situations like these. 😄